### DISCUSSION TITLE
problem in filters in results after scanning IP randoms

### DISCUSSION THREAD

**Discussion by haqpentest** [cea-sec/ivre#1543] [Q&A] [answered]
Now, I tried to classify devices as webcam/telephone/network in Fun button in GUI but there isn't anything.I installed IVRE by docker following docs. Can you show me solutions?
Thank you very much

**Comment by p-l-** [comment, marked as answer]
The filters are `devicetype:` filters, which use the service devicetype data found by Nmap when using service scans. So you have no results because either you don't have devices identified by Nmap as webcam/phone/whatever or because you did not use the correct Nmap options (please refer to Nmap documentation).

**Reply by haqpentest** [reply]
Thank you very much sir. I want to ask more.
I am building agent following documentation. When to the step "ivre runscans --output Agent > agent; chmod +x agent" that generating script file. In the file, I can see the script to scan
![image](https://github.com/ivre/ivre/assets/136410921/859034f8-5fd4-47b3-a96d-e887ed20a426)
when I try to scan IP random, I "ve had results in GUI but not to classify devicetype as webcam/phone...
My question is that by using agent following documentation, IVRE is possible to scan devicetype, isn't it? If it is possible, how do I fix??(I tried to use some guidance in Nmap doc but it is not successful).
I am looking forward to hearing from you.
Thank you very much

**Reply by p-l-** [reply]
You need to adapt the Nmap configuration used by IVRE (as described in [the documentation](https://doc.ivre.rocks/en/latest/install/config.html#nmap-scan-templates)).
Based on the script you provide, you use `-A` which is a shortcut for many options and the device type should be detected.
So you need to understand why Nmap does not recognize the device type, and that's an Nmap question, not an IVRE question.
Happy scanning!

**Reply by haqpentest** [reply]
Thank you sir very much. It means that the default script file ,which is generated in building agent following documentation ,can't scan devicetype, it need to be configured. I used to add some parameters but there is not anything, Can you suggest me any solution if you have free time??

**Reply by p-l-** [reply]
No. The default script can scan device type. For some reasons, Nmap won't find device type against your targets, and I have no idea why.

**Reply by haqpentest** [reply]
Can you give me some IP address list to test, sir?? Because I used IP random so I suppose that it is problem.
THis is my script file that is genrerated automatically .I don't realize problem in file that can't detect device type. Can you help me, please?

[agent.txt](https://github.com/ivre/ivre/files/11858918/agent.txt)

**Reply by p-l-** [reply]
For god sake this is an Nmap problem, not an IVRE problem. Now ask your questions to Nmap support.
